src/components/NotificationsDrawer/DrawerPanelContent.tsx

```diff
@@ -39,19 +57,170 @@ const EmptyNotifications = () => (
+  };
+
+  const onMarkAllAsUnread = () => {
+    dispatch(markAllNotificationsAsUnread());
+    setIsDropdownOpen(false);
+  };
+
+  const onFilterSelect = (chosenFilter: string) => {
+    activeFilters.includes(chosenFilter)
+      ? setActiveFilters(activeFilters.filter((filter) => filter !== chosenFilter))
+      : setActiveFilters([...activeFilters, chosenFilter]);
+  };
+
+  const dropdownItems = [
+    <DropdownItem key="read all" onClick={onMarkAllAsRead} isDisabled={notifications.length === 0}>
+      Mark visible as read
+    </DropdownItem>,
+    <DropdownItem key="unread all" onClick={onMarkAllAsUnread} isDisabled={notifications.length === 0}>
+      Mark visible as unread
+    </DropdownItem>,
+    <Divider key="divider" />,
+    <DropdownItem key="event log" onClick={() => navigate('/settings/notifications/eventlog')}>
+      <Flex>
+        <FlexItem>View event log</FlexItem>
+        <FlexItem align={{ default: 'alignRight' }}>
+          <Icon className="pf-v5-u-ml-auto">
+            <ExternalLinkAltIcon />
+          </Icon>
+        </FlexItem>
+      </Flex>
+    </DropdownItem>,
+    isOrgAdmin && (
+      <DropdownItem key="notification settings" onClick={() => navigate('/settings/notifications/configure-events')}>
+        <Flex>
+          <FlexItem>Configure notification settings</FlexItem>
+          <FlexItem align={{ default: 'alignRight' }}>
+            <Icon className="pf-v5-u-ml-auto">
+              <ExternalLinkAltIcon />
+            </Icon>
+          </FlexItem>
+        </Flex>
+      </DropdownItem>
+    ),
+    <DropdownItem key="notification preferences" onClick={() => navigate('/settings/notifications/user-preferences')}>
+      <Flex>
+        <FlexItem>Manage my notification preferences</FlexItem>
+        <FlexItem align={{ default: 'alignRight' }}>
+          <Icon className="pf-v5-u-ml-auto">
+            <ExternalLinkAltIcon />
+          </Icon>
+        </FlexItem>
+      </Flex>
+    </DropdownItem>,
+  ];
+
+  const filterDropdownItems = () => {
+    return [
+      <DropdownGroup key="filter-label" label="Show notifications for...">
+        {filterConfig.map((source) => (
+          <DropdownItem key={source.value} onClick={() => onFilterSelect(source.value)}>
+            <Checkbox isChecked={activeFilters.includes(source.value)} id={source.value} className="pf-v5-u-mr-sm" />
+            {source.title}
+          </DropdownItem>
+        ))}
+        <Divider />
+        <DropdownItem key="reset-filters" onClick={() => setActiveFilters([])}>
+          <Button variant="link" isDisabled={activeFilters.length === 0} isInline>
+            Reset filters
+          </Button>
+        </DropdownItem>
+      </DropdownGroup>,
+    ];
+  };
+
+  const renderNotifications = () => {
+    if (notifications.length === 0) {
+      return <EmptyNotifications />;
+    }
+
+    const sortedNotifications = orderBy(
+      filteredNotifications?.length > 0 ? filteredNotifications : notifications,
```

Comment:
This is not ideal, if user selects a complex enough filter they will see every notification, because the `filteredNotifications` will be empty. Instead of `filteredNotifications` we should check if `activeFilters` is empty or not.

Reply:
Great catch, thanks. Now displaying nothing when active filter has no matching notifications.